assets/js/data/checkout/reducers.ts

```diff
@@ -0,0 +1,185 @@
+					? {
+							...state,
+							status: STATUS.IDLE,
+					  }
+					: state;
+			break;
+
+		case types.SET_REDIRECT_URL:
+			newState =
+				action.redirectUrl !== undefined &&
+				action.redirectUrl !== state.redirectUrl
+					? {
+							...state,
+							redirectUrl: action.redirectUrl,
+					  }
+					: state;
+			break;
+
+		case types.SET_PROCESSING_RESPONSE:
+			newState = {
+				...state,
+				processingResponse: action.data as PaymentResult,
+			};
+			break;
+
+		case types.SET_COMPLETE:
+			newState = {
+				...state,
+				status: STATUS.COMPLETE,
+				redirectUrl:
+					typeof action.data?.redirectUrl === 'string'
+						? action.data.redirectUrl
+						: state.redirectUrl,
+			};
+			break;
+		case types.SET_PROCESSING:
+			newState = {
+				...state,
+				status: STATUS.PROCESSING,
+				hasError: false,
+			};
+			break;
+
+		case types.SET_BEFORE_PROCESSING:
+			newState = {
+				...state,
+				status: STATUS.BEFORE_PROCESSING,
+				hasError: false,
+			};
+			break;
+
+		case types.SET_AFTER_PROCESSING:
+			newState = {
+				...state,
+				status: STATUS.AFTER_PROCESSING,
+			};
+			break;
+
+		case types.SET_HAS_ERROR:
+			newState = {
+				...state,
+				hasError: action.hasError,
+				status:
+					state.status === STATUS.PROCESSING ||
+					state.status === STATUS.BEFORE_PROCESSING
+						? STATUS.IDLE
+						: state.status,
+			};
+			break;
+
+		case types.INCREMENT_CALCULATING:
+			newState = {
+				...state,
+				calculatingCount: state.calculatingCount + 1,
+			};
+			break;
+
+		case types.DECREMENT_CALCULATING:
+			newState = {
+				...state,
+				calculatingCount: Math.max( 0, state.calculatingCount - 1 ),
+			};
+			break;
+
+		case types.SET_CUSTOMER_ID:
+			if ( action.customerId !== undefined ) {
+				newState = {
+					...state,
+					customerId: action.customerId,
+				};
+			}
+			break;
+
+		case types.SET_ORDER_ID:
+			if ( action.orderId !== undefined ) {
+				newState = {
+					...state,
+					orderId: action.orderId,
+				};
+			}
+			break;
+
+		case types.SET_SHIPPING_ADDRESS_AS_BILLING_ADDRESS:
+			if (
+				action.useShippingAsBilling !== undefined &&
+				action.useShippingAsBilling !== state.useShippingAsBilling
+			) {
+				newState = {
+					...state,
+					useShippingAsBilling: action.useShippingAsBilling,
+				};
+			}
+			break;
+
+		case types.SET_SHOULD_CREATE_ACCOUNT:
+			if (
+				action.shouldCreateAccount !== undefined &&
+				action.shouldCreateAccount !== state.shouldCreateAccount
+			) {
+				newState = {
+					...state,
+					shouldCreateAccount: action.shouldCreateAccount,
+				};
+			}
+			break;
+
+		case types.SET_ORDER_NOTES:
+			if (
+				action.orderNotes !== undefined &&
+				state.orderNotes !== action.orderNotes
+			) {
+				newState = {
+					...state,
+					orderNotes: action.orderNotes,
+				};
+			}
+			break;
+
+		case types.SET_EXTENSION_DATA:
+			if (
+				action.extensionData !== undefined &&
+				state.extensionData !== action.extensionData
```

Comment:
`state.extensionData` and `action.extensionData` will never be `===` because the action takes a brand new object when it's dispatched, doesn't it?

Reply:
Yeah good spot, I think that's a bug comparing these objects by reference, we should be comparing by value to see if any props are different. This is happening for quite a few of the actions here, so I think we should fix this separately after merging this PR to not cause too many uknown. I've created an issue [here](https://github.com/woocommerce/woocommerce-blocks/issues/6506)